From fa849d6d4c532ceb1bdaee777bc3f708f967e369 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Tue, 23 Apr 2019 03:36:36 +0000
Subject: [PATCH 1348/1418] drm/i915/gvt: disable the preemption timeout in
 GVT-g guest

In GVT-g guest, preemption is disabled. However, the force preemption
related patches will trigger force GPU reset, which impacts the guest
QoS. So disabled it in this patch for GVT-g guests.

Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/intel_lrc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index 2a6b9fb..dee9c7e 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -1231,6 +1231,7 @@ static void __update_queue(struct intel_engine_cs *engine,
 
 	/* Set a timer to force preemption vs hostile userspace */
 	if (timeout &&
+	    !intel_vgpu_active(engine->i915) &&
 	    __execlists_need_preempt(prio, execlists->queue_priority)) {
 		GEM_TRACE("%s preempt timeout=%uns\n", engine->name, timeout);
 
-- 
2.7.4

